pkgs/applications/emulators/ppsspp/default.nix

```diff
@@ -3,76 +3,119 @@
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "ppsspp"
+      + lib.optionalString enableQt "-qt"
+      + lib.optionalString (!enableQt) "-sdl"
+      + lib.optionalString forceWayland "-wayland";
+    version = "1.13.1";
 
-  postPatch = ''
-    substituteInPlace git-version.cmake --replace unknown ${finalAttrs.src.rev}
-    substituteInPlace UI/NativeApp.cpp --replace /usr/share $out/share
-  '';
+    src = fetchFromGitHub {
+      owner = "hrydgard";
+      repo = "ppsspp";
+      rev = "v${finalAttrs.version}";
+      fetchSubmodules = true;
+      sha256 = "sha256-WsFy2aSOmkII2Lte5et4W6qj0AXUKWWkYe88T0OQP08=";
+    };
 
-  nativeBuildInputs = [
-    cmake
-    pkg-config
-    python3
-    wrapQtAppsHook
-  ];
+    postPatch = ''
+      substituteInPlace git-version.cmake --replace unknown ${finalAttrs.src.rev}
+      substituteInPlace UI/NativeApp.cpp --replace /usr/share $out/share
+    '';
 
-  buildInputs = [
-    SDL2
-    ffmpeg
-    glew
-    libzip
-    qtbase
-    qtmultimedia
-    snappy
-    zlib
-  ];
+    nativeBuildInputs = [
+      cmake
+      copyDesktopItems
+      makeWrapper
+      pkg-config
+      python3
+      wrapQtAppsHook
+    ];
 
-  cmakeFlags = [
-    "-DHEADLESS=OFF"
-    "-DOpenGL_GL_PREFERENCE=GLVND"
-    "-DUSE_SYSTEM_FFMPEG=ON"
-    "-DUSE_SYSTEM_LIBZIP=ON"
-    "-DUSE_SYSTEM_SNAPPY=ON"
-    "-DUSING_QT_UI=ON"
-  ];
+    buildInputs = [
+      SDL2
+      ffmpeg
+      (glew.override { enableEGL = forceWayland; })
+      libzip
+      qtbase
+      qtmultimedia
+      snappy
+      zlib
+    ] ++ lib.optional enableVulkan vulkan-loader
+      ++ lib.optionals vulkanWayland [ wayland libffi ];
 
-  installPhase = ''
-    runHook preInstall
-    mkdir -p $out/share/ppsspp
-    install -Dm555 PPSSPPQt $out/bin/ppsspp
-    mv assets $out/share/ppsspp
-    runHook postInstall
-  '';
+    cmakeFlags = [
+      "-DHEADLESS=${if enableQt then "OFF" else "ON"}"
+      "-DOpenGL_GL_PREFERENCE=GLVND"
+      "-DUSE_SYSTEM_FFMPEG=ON"
+      "-DUSE_SYSTEM_LIBZIP=ON"
+      "-DUSE_SYSTEM_SNAPPY=ON"
+      "-DUSE_WAYLAND_WSI=${if vulkanWayland then "ON" else "OFF"}"
+      "-DUSING_QT_UI=${if enableQt then "ON" else "OFF"}"
+    ];
 
-  meta = with lib; {
-    homepage = "https://www.ppsspp.org/";
-    description = "A HLE Playstation Portable emulator, written in C++";
-    license = licenses.gpl2Plus;
-    maintainers = with maintainers; [ AndersonTorres ];
-    platforms = platforms.linux;
-  };
-})
-# TODO: add SDL headless port
+    desktopItems = [(makeDesktopItem {
+      desktopName = "PPSSPP";
+      name = "ppsspp";
+      exec = "ppsspp";
+      icon = "ppsspp";
+      comment = "Play PSP games on your computer";
+      categories = [ "Game" "Emulator" ];
+    })];
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out/share/{applications,ppsspp}
+    '' + (if enableQt then ''
+      install -Dm555 PPSSPPQt $out/bin/ppsspp
+      wrapProgram $out/bin/ppsspp \
```

Comment:
What are the parameters of this wrapping?

Reply:
In case `enableVulkan` is overridden to be `true`, it would be wrapped with `--prefix LD_LIBRARY_PATH : ${vulkanPath}`. This should allow the Qt front end to detect Vulkan, though not able to use it for now.

Reply:
Ugly as hell.